hazelcast/src/main/java/com/hazelcast/cluster/impl/ClusterServiceImpl.java

```diff
@@ -120,93 +122,112 @@
     private volatile long lastHeartBeat;
 
-    private long firstJoinRequest;
-
-    private final ConcurrentMap<MemberImpl, Long> heartbeatTimes = new ConcurrentHashMap<MemberImpl, Long>();
-
-    private final ConcurrentMap<MemberImpl, Long> masterConfirmationTimes = new ConcurrentHashMap<MemberImpl, Long>();
-
-    private final ExceptionHandler whileFinalizeJoinsExceptionHandler;
-
-    private final ClusterClockImpl clusterClock;
-
-    private String clusterId = null;
-
-    public ClusterServiceImpl(final Node node) {
+    public ClusterServiceImpl(Node node) {
         this.node = node;
         nodeEngine = node.nodeEngine;
+
         logger = node.getLogger(ClusterService.class.getName());
         clusterClock = new ClusterClockImpl(logger);
-        whileFinalizeJoinsExceptionHandler =
-                logAllExceptions(logger, "While waiting finalize join calls...", Level.WARNING);
+        whileFinalizeJoinsExceptionHandler = logAllExceptions(logger, "While waiting finalize join calls...", Level.WARNING);
+
         thisAddress = node.getThisAddress();
         thisMember = node.getLocalMember();
+
         setMembers(thisMember);
         sendMembershipEvents(Collections.<MemberImpl>emptySet(), Collections.singleton(thisMember));
-        waitMillisBeforeJoin = node.groupProperties.WAIT_SECONDS_BEFORE_JOIN.getInteger() * 1000L;
-        maxWaitMillisBeforeJoin = node.groupProperties.MAX_WAIT_SECONDS_BEFORE_JOIN.getInteger() * 1000L;
-        maxNoHeartbeatMillis = node.groupProperties.MAX_NO_HEARTBEAT_SECONDS.getInteger() * 1000L;
-        maxNoMasterConfirmationMillis = node.groupProperties.MAX_NO_MASTER_CONFIRMATION_SECONDS.getInteger() * 1000L;
+
+        maxWaitMillisBeforeJoin = getAsMilliSeconds(node.groupProperties.MAX_WAIT_SECONDS_BEFORE_JOIN);
+        waitMillisBeforeJoin = getAsMilliSeconds(node.groupProperties.WAIT_SECONDS_BEFORE_JOIN);
+        maxNoHeartbeatMillis = getAsMilliSeconds(node.groupProperties.MAX_NO_HEARTBEAT_SECONDS);
+        maxNoMasterConfirmationMillis = getAsMilliSeconds(node.groupProperties.MAX_NO_MASTER_CONFIRMATION_SECONDS);
+
+        heartbeatIntervalMillis = getHeartBeatInterval(node.groupProperties.HEARTBEAT_INTERVAL_SECONDS);
+        pingIntervalMillis = heartbeatIntervalMillis * HEART_BEAT_INTERVAL_FACTOR;
+
         icmpEnabled = node.groupProperties.ICMP_ENABLED.getBoolean();
         icmpTtl = node.groupProperties.ICMP_TTL.getInteger();
         icmpTimeout = node.groupProperties.ICMP_TIMEOUT.getInteger();
-        long heartbeatInterval = node.groupProperties.HEARTBEAT_INTERVAL_SECONDS.getInteger();
-        heartbeatIntervalMillis = heartbeatInterval > 0 ? heartbeatInterval * 1000L : 1000L;
-        pingIntervalMillis = heartbeatIntervalMillis * 10;
+
         node.connectionManager.addConnectionListener(this);
 
         registerMetrics();
     }
 
-    void registerMetrics() {
+    private static long getAsMilliSeconds(GroupProperties.GroupProperty groupProperty) {
```

Comment:
Methods like this maybe a candidate for a Util class, but I didn't pull out any method from this class during this refactoring step.